rfd/xxxx-config-resource-docs.md

```diff
@@ -0,0 +1,554 @@
+
+{/*Automatically generated from: api/types/types.pb.go*/}
+
+|Field Name|Description|Type|
+|---|---|---|
+|`name`|The http header name.|`string`|
+|`value`|The http header value.|`string`|
+
+\`\`\`yaml
+name: string
+value: string
+\`\`\`
+```
+
+### Working with parsed Go source
+
+The generator will use
+[`packages.Load`](https://pkg.go.dev/golang.org/x/tools/go/packages#Load) to
+load all the packages named in generator's configuration file and return a
+`[]*packages.Package`. It then recursively traverses each `packages.Package`'s
+`Imports` map, adding to a final slice of parsed packages.
+
+The generator declares two maps. The key for each map, `declKey`, includes the
+package path and named type of a declaration:
+
+- Declaration map: `map[declKey]sourceDecl`: Looking up AST data for a given
+  declaration in order to build template data. A `sourceDecl` is a struct with
+  the following definition:
+
+  ```go
+  type sourceDecl struct {
+   decl  *ast.Decl // Declaration data
+   file  string   // File path where the declaration was made
+  }
+  ```
+
+- Final data map: `map[declKey]Resource`: Tracking type data to use to populate
+  a template. This ensures that each declaration is only included once.
+
+In each `*packages.Package` in the slice mentioned earlier, the generator ranges
+through the package's `Syntax` field, which is an `[]*ast.File`. Within each
+`*ast.File`, the generator ranges through the file's
+[`Decls`](https://pkg.go.dev/go/ast#File) field and adds each declaration to the
+declaration map. The `CompiledGoFiles` field in `packages.Package` lists paths
+of Go source files in the same order as the `Syntax` field, so we can use this
+to populate the `file` property of each `sourceDecl` in the map.
+
+The next step is to look up each user-configured struct type within the
+declaration map and begin generating a `Resource` for that struct type, plus
+additional `Resource` structs for the types of its fields (and their types
+and so on). The sequence is:
+
+- Look up a user-defined struct type from the configuration within the
+  declaration map.
+- Construct a `Resource` from the struct type and insert it into the final data
+  map (unless a `Resource` already exists there).
+- Process the fields of the user-defined struct type using the rules described
+  [below](#processing-field-types).
+
+Once this is done, the generator will convert the  `map[declKey]Resource` into a
+single `[]Resource` to feed into the template.
+
+#### Alternative: Using analysis drivers
+
+The Go Tools repository provides the `multichecker`
+(https://pkg.go.dev/golang.org/x/tools/go/analysis/multichecker) and
+`singlechecker`
+(https://pkg.go.dev/golang.org/x/tools/go/analysis/singlechecker) packages
+within `golang.org/x/tools/go/analysis`, which run custom static analysis
+functions, `analysis.Analyzer`s. However, these static analysis tools are
+intended for linting, not generating reference documentation. 
```

Comment:
> However, these static analysis tools are intended for linting, not generating reference documentation

Design intention is key. It might work for a while, but it seems poor form to depend on it.

A program as simpler as we can make it, without unnecessary attachments, seems better long-term.